### ISSUE TITLE
Allow total to be changed at runtime.

### ISSUE THREAD

**Issue by bramp** [RohitEdathil/ConsoleBars#5]
I have a progress bar where the total may change over time.

I propose a update to the API like so:

```dart
void update(int n) { ... }
```

to
```dart
void update(int n, {int? total}) { ... }
```

and then if I call `update(1, total: 100)` the total gets updated appropriately.

Alternatively there could be a new method to update the total, but it seems reasonable to always update the current progress and total at the same time.

**Comment by RohitEdathil** [comment]
That sounds like a good idea @bramp. On top of that, maybe in some use cases, developers may want to change the description midway (like from `Unzipping` to `Copying Files`). 

I think the solution is to define getters and setters on `total` and `desc` properties. The setters will automatically trigger a render call. I am leaving the update method alone for now as people may not want to manually set the progress every time they want to update the details of the progress bar.

**Comment by RohitEdathil** [comment]
Just published `1.2.0` with the setters ready. Hope it works for you.

**Comment by bramp** [comment]
Thanks!
